src/Spectre.libGaClassifier/GaFitnessFunction.h

```diff
@@ -0,0 +1,64 @@
+/*
+* GaFitnessFunction.h
+* It is fitness function for any classifier for GaClassifier.
+*
+Copyright 2018 Spectre Team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+#include "Spectre.libGenetic/FitnessFunction.h"
+#include "Spectre.libClassifier/IClassifier.h"
+#include <memory>
+#include "Spectre.libClassifier/SplittedOpevCvDataset.h"
+#include "Spectre.libClassifier/ConfusionMatrix.h"
+
+namespace spectre::supervised
+{
+/// <summary>
+/// It is fitness function for any classifier for GaClassifier.
+/// </summary>
+class GaFitnessFunction : public spectre::algorithm::genetic::FitnessFunction
+{
+public:
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GaFitnessFunction"/> class.
+    /// </summary>
+    /// <param name="classifier">The classifier.</param>
+    /// <param name="data">The splitted data.</param>
+    GaFitnessFunction(IClassifier& classifier, SplittedOpenCvDataset& data);
```

Comment:
If we start running evaluations of `GaFitnessFunction` for `IClassifier&`, we are operating on single instance. Imagine, how much it blows up in parallel :)

We already have it called in parallel!

I'd consider consuming the instance here (probably as `std::unique_ptr<IClassifier>`) or just remembering it (as `const IClassifier&`), and then copying it for each evaluation. This should be lightweight, since this classifier before fitting should consist of the parameterization only, not a model itself. @Sand3r-, what do you think about it?

Reply:
done